lib/vagrant-proxyconf/action/configure_docker_proxy.rb

```diff
@@ -66,8 +67,21 @@ def detect_export
           end
         end
 
+        def create_sysytemd_config
+          dir = '/etc/systemd/system/docker.service.d/'
+          file_name = 'http-proxy.conf'
+          path = dir + file_name
+
+          @machine.communicate.tap do |comm|
+            comm.sudo("mkdir -m 755 -p #{dir}")
+            local_tmp = tempfile(systemd_config)
+            comm.upload(local_tmp.path, path)
```

Comment:
Hi, please note, that this didn't work for me, when I tested it to "vagrant up" the current fabric8-installer on OpenShift. The error was: 

```
Failed to upload a file to the guest VM via SCP due to a permissions
error. This is normally because the SSH user doesn't have permission
to write to the destination location. Alternately, the user running
Vagrant on the host machine may not have permission to read the file.

Source: C:/Users/TOTTO~1.INF/AppData/Local/Temp/vagrant20160829-11648-biuxz8
Dest: /etc/systemd/system/docker.service.d/http-proxy.conf
```

So I changed it like this (to copy the file to temp first, then change owner and file path):
```
          tmp = "/tmp/vagrant-proxyconf-dockerproxy" 	

...

          @machine.communicate.tap do |comm|
             comm.sudo("mkdir -m 755 -p #{dir}")
             local_tmp = tempfile(systemd_config)
             comm.upload(local_tmp.path, tmp)
             comm.sudo("chmod 0644 #{tmp}")
             comm.sudo("chown root:root #{tmp}")
             comm.sudo("mv -f #{tmp} #{path}")			              
           end
```